packages/liveblocks-react/src/room.tsx

```diff
@@ -797,6 +799,63 @@ function useReportTextEditor(editor: TextEditorType, rootKey: string): void {
   }, [room, editor, rootKey]);
 }
 
+/** @private - Internal API, do not rely on it. */
+function useYjsProvider(): IYjsProvider | undefined {
+  const room = useRoom();
+
+  const subscribe = React.useCallback(
+    (onStoreChange: () => void): UnsubscribeCallback => {
+      return room[kInternal].yjsProviderDidChange.subscribe(onStoreChange);
+    },
+    [room]
+  );
+
+  const getSnapshot = React.useCallback((): IYjsProvider | undefined => {
+    return room[kInternal].getYjsProvider();
+  }, [room]);
+
+  return useSyncExternalStore(subscribe, getSnapshot, getSnapshot);
+}
+
+/** @private - Internal API, do not rely on it. */
+function useCreateTextMention(): (userId: string, mentionId: string) => void {
+  const room = useRoom();
+  return React.useCallback(
+    (userId: string, mentionId: string): void => {
+      room[kInternal]
+        .createTextMention(userId, mentionId)
+        .catch((err): void => {
+          console.error(
+            `Cannot create text mention for user '${userId}' and mention '${mentionId}'`,
+            err
+          );
+        });
+    },
+    [room]
+  );
+}
+
+/** @private - Internal API, do not rely on it. */
+function useDeleteTextMention(): (mentionId: string) => void {
+  const room = useRoom();
+  return React.useCallback(
+    (mentionId: string): void => {
+      room[kInternal].deleteTextMention(mentionId).catch((err): void => {
+        console.error(`Cannot delete text mention '${mentionId}'`, err);
+      });
+    },
+    [room]
+  );
+}
+
+/** @private - Internal API, do not rely on it. */
+function useIsResolveMentionSuggestionsDefined(): boolean {
+  const client = useClient();
+  return React.useMemo((): boolean => {
+    return client[kInternal].resolveMentionSuggestions !== undefined;
+  }, [client]);
```

Comment:
We should only use `useMemo` for computationally-expensive expressions. In this case, you're just doing simple member access, so just doing this should be enough:

```suggestion
  return client[kInternal].resolveMentionSuggestions !== undefined;
```

Reply:
Additionally, this hook seems a bit over-specific. Should we expose this as `useResolveMentionSuggestions` instead, and doing the `!== undefined` part in the call site? That way the hook seems a bit more composable/flexible to me, and encourages DRYness.

There are other parts of the code base that already access the `resolveMentionSuggestions` prop ([for example](https://github.com/liveblocks/liveblocks/blob/8ecfd777f7995592187f484bc7b4f7ffb5c0a5a5/packages/liveblocks-react/src/use-mention-suggestions.ts#L24)), maybe they can benefit from the same, more generic, hook?

Reply:
@nvie can you PTAL? Thanks 🙏🏻

Reply:
Ping @nvie when you get a chance 🙂 ☝🏻

Reply:
(Sorry I missed these notifications @SugarDarius! Was just going through all my unread mail today and saw I missed these pings — so sorry about that! GitHub is sending me too many emails, I need to improve my notification settings.)

Reply:
@nvie no worries 🙂